### PR TITLE
PMM-12806 VictoriaMetrics to give preference to environment vars

### PR DESCRIPTION
**Pull request by ademidoff**
PMM-12806

Link to the Feature Build: SUBMODULES-3515

- Links to related pull requests (optional)

### CODE REVIEW

File: managed/services/supervisord/supervisord_test.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ func TestConfig(t *testing.T) {
 	}
 	settings.VictoriaMetrics.CacheEnabled = false
 
+	// Test a couple of environment variables being picked up by VictoriaMetrics.
+	err = os.Setenv("VM_search_maxQueryLen", "2MB")
```
[the remark this answers is not in the document]
Comment (ademidoff) [reply managed/services/supervisord/supervisord_test.go]: t.Setenv() is incompatible with t.Parallel()

File: managed/services/supervisord/supervisord.go [hunk 2]
```diff
@@ -419,23 +428,40 @@ func (s *Service) marshalConfig(tmpl *template.Template, settings *models.Settin
...
-		"ClickhouseAddr":           clickhouseAddr,
-		"ClickhouseDataSourceAddr": clickhouseDataSourceAddr,
-		"ClickhouseDatabase":       clickhouseDatabase,
-		"ClickhousePoolSize":       clickhousePoolSize,
-		"ClickhouseBlockSize":      clickhouseBlockSize,
-		"ClickhouseHost":           clickhouseAddrPair[0],
-		"ClickhousePort":           clickhouseAddrPair[1],
+		"DataRetentionHours":           int(settings.DataRetention.Hours()),
+		"DataRetentionDays":            int(settings.DataRetention.Hours() / 24),
+		"VMAlertFlags":                 s.vmParams.VMAlertFlags,
```
Comment (JiriCtvrtka) [inline managed/services/supervisord/supervisord.go:444]: Could be vmParams nil?

File: managed/services/supervisord/supervisord.go [hunk 3]
```diff
@@ -419,23 +428,40 @@ func (s *Service) marshalConfig(tmpl *template.Template, settings *models.Settin
 	clickhousePoolSize := getValueFromENV("PERCONA_TEST_PMM_CLICKHOUSE_POOL_SIZE", "")
 	clickhouseBlockSize := getValueFromENV("PERCONA_TEST_PMM_CLICKHOUSE_BLOCK_SIZE", "")
 	clickhouseAddrPair := strings.SplitN(clickhouseAddr, ":", 2)
+	vmSearchDisableCache := getValueFromENV("VM_search_disableCache", strconv.FormatBool(!settings.VictoriaMetrics.CacheEnabled))
```
Comment (JiriCtvrtka) [inline managed/services/supervisord/supervisord.go:431]: Any reason why env vars are not uppercase in case of VM_ prefix?

**Review by JiriCtvrtka — APPROVED** [review]
LGTM, just two questions.

File: managed/services/supervisord/supervisord.go (hunk 3 above)
Comment (ademidoff) [reply managed/services/supervisord/supervisord.go:431]: Yes, the naming is opinionated and comes from VM command line parameters. They have adapted those to be envvars as well, hence the weird naming.

Comment (ademidoff) [reply managed/services/supervisord/supervisord.go:431]: You can check the docs:
- https://docs.victoriametrics.com/Single-server-VictoriaMetrics.html#list-of-command-line-flags
- https://docs.victoriametrics.com/Single-server-VictoriaMetrics.html#environment-variables

**Review by idoqo — APPROVED** [review]

File: managed/services/supervisord/supervisord_test.go [hunk 4]
```diff
@@ -55,6 +55,13 @@ func TestConfig(t *testing.T) {
 	}
 	settings.VictoriaMetrics.CacheEnabled = false
 
+	// Test a couple of environment variables being picked up by VictoriaMetrics.
+	// NOTE: we can't use t.Setenv() here because of parallel tests.
+	err = os.Setenv("VM_search_maxQueryLen", "2MB") //nolint:tenv
+	require.NoError(t, err)
+	err = os.Setenv("VM_search_latencyOffset", "10s") //nolint:tenv
+	require.NoError(t, err)
+
```
[the remark this answers is not in the document]
Comment (ademidoff) [reply managed/services/supervisord/supervisord_test.go:64]: The tests will pass either way, but you are right - for the sake of clarity and maintainability it's better to make a separate one just for the envvars.

[the remark this answers is not in the document]
Comment (ademidoff) [reply managed/services/supervisord/supervisord_test.go]: Fixed.

**Pull request merged**